src/kernel/arch/x86/interrupts.zig

```diff
@@ -40,7 +41,8 @@ export fn commonStub() callconv(.Naked) void {
         \\mov   %%esp, %%eax
         \\push  %%eax
         \\call  handler
-        \\pop   %%eax
+        \\mov   %%eax, %%esp
```

Comment:
After my experiences with how the release modes can change the ABI, I was wondering if it would be better to delegate the `push eax; call handler; mov eax, esp` to Zig, with something like.
```
const esp = handler(asm ("mov %[stack], %%esp" : [stack] "=r" (-> usize));
asm volatile ("mov %%esp, %[stack]" :: [stack] "r" (esp));
```
That way, this code will work even when a different ABI is being used, due to optimisations.

Reply:
I have had ago at this, but always get a page fault. When I go back to the old (original) version, it works. Could you see if you are able to get this working.

Reply:
Sure, I'll take a look at the assembly too

Reply:
Yeah doing this adds a load of gunk in debug mode that screws with the stack. Let's keep it like it is at the moment but be aware that we may need to pay attention to the ABI (passing the argument and receiving the return value differently) if we run into problems later.

Reply:
It adds
```
mov    %esp,%esi
c011bd86:       8b 7e 34                mov    0x34(%esi),%edi
c011bd89:       83 ff 20                cmp    $0x20,%edi
c011bd8c:       72 20                   jb     c011bdae <commonStub+0x3e>
c011bd8e:       81 ff 80 00 00 00       cmp    $0x80,%edi
c011bd94:       74 18                   je     c011bdae <commonStub+0x3e>
c011bd96:       56                      push   %esi
c011bd97:       e8 54 f2 ff ff          call   c011aff0 <irqHandler>
c011bd9c:       83 c4 04                add    $0x4,%esp
c011bd9f:       89 c4                   mov    %eax,%esp
```
in release-fast and release-small, and

```
c013af74:       89 e0                   mov    %esp,%eax
c013af76:       89 45 fc                mov    %eax,-0x4(%ebp)
c013af79:       8b 45 fc                mov    -0x4(%ebp),%eax
c013af7c:       83 f8 00                cmp    $0x0,%eax
c013af7f:       89 45 f4                mov    %eax,-0xc(%ebp)
c013af82:       75 0d                   jne    c013af91 <commonStub+0x31>
c013af84:       31 d2                   xor    %edx,%edx
c013af86:       8d 0d f4 e5 13 c0       lea    0xc013e5f4,%ecx
c013af8c:       e8 6f 80 fc ff          call   c0103000 <panic>
c013af91:       8b 45 f4                mov    -0xc(%ebp),%eax
c013af94:       83 e0 00                and    $0x0,%eax
c013af97:       83 f8 00                cmp    $0x0,%eax
c013af9a:       74 0d                   je     c013afa9 <commonStub+0x49>
c013af9c:       31 d2                   xor    %edx,%edx
c013af9e:       8d 0d fc e5 13 c0       lea    0xc013e5fc,%ecx
c013afa4:       e8 57 80 fc ff          call   c0103000 <panic>
c013afa9:       8b 45 f4                mov    -0xc(%ebp),%eax
c013afac:       89 04 24                mov    %eax,(%esp)
c013afaf:       e8 4c ff ff ff          call   c013af00 <handler>
c013afb4:       89 45 f8                mov    %eax,-0x8(%ebp)
c013afb7:       8b 45 f8                mov    -0x8(%ebp),%eax
```

in debug mode thanks to the safety checks and lack of optimisations

Reply:
yeh, I was expecting this to happen. Will just have to keep an eye on it